Created ShopX-React Tool

v-sokolov opened the pull request:
Appending / Removing items in the main catalog based by design shopX web-site

Commit:
Created ShopX-React Tool

Appending / Removing items in the main catalog

Commit:
Minor Codacy fixes

Commit:
Minor Codacy fixes css

muha2399 commented on ReactApps/reacttest/src/App.css:
```diff
@@ -0,0 +1,220 @@
+.App {
```
Please split styles into different files for each component. Example : App.js - App.css, Wrapper.js - Wrapper.cs, Card.js- Card.css.

muha2399 commented on ReactApps/reacttest/src/card.js:
```diff
@@ -0,0 +1,59 @@
+import React, {Component} from "react";
+
+class ItemCard extends Component {
```
Do we really need state here ? It never changes.

muha2399 commented on ReactApps/reacttest/src/card.js:
```diff
@@ -0,0 +1,59 @@
+                <div className="sectionBody">
+                    <div className="text">
+                        <div>
+                            <p className="itemTitle">{this.props.title}</p>
```
You can destructure props {title, oldPrice, newPrice} = this.props before return and use it.

muha2399 commented on ReactApps/reacttest/src/wrapper.jsx:
```diff
@@ -0,0 +1,59 @@
+        super(props);
+        this.id = 0;
+        this.state = {
+            data: [
```
Do we need this initial state if its empty ? Please remove comments.

v-sokolov replied:
yes, it's need to render empty catalog

muha2399 commented on ReactApps/reacttest/src/wrapper.jsx:
```diff
@@ -0,0 +1,59 @@
+            // <ItemCard title="LavaVtase" oldPrice="68" newPrice ="35"/>
+            ]
+        };
+        //Бинд ивэнта клик к этому компоненту
```
Please use English for commenting.

muha2399 commented on ReactApps/reacttest/src/wrapper.jsx:
```diff
@@ -0,0 +1,59 @@
+                    <button id="add" onClick={this.handleClick}>Append</button>
+                </div>
+                <div className="catalog">
+                    {this.state.data}
```
Data should be raw data, not a component. So here you should create component for each data item in state.

Commit:
Fixes after muha2399 review

All styles divided by different files
Card's props were destructured
Data reworked by component to Raw-Data
Unnecessary comments were deleted

Commit:
.gitignore

Commit:
Updated git ignore
